crypto/src/main/java/org/hyperledger/besu/crypto/Signature.java

```diff
@@ -0,0 +1,136 @@
+package org.hyperledger.besu.crypto;
+
+import com.google.common.base.Suppliers;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.MutableBytes;
+import org.apache.tuweni.units.bigints.UInt256;
+
+import java.math.BigInteger;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class Signature {
```

Comment:
Probably should rename it something more specific, like `SECPSignature`.

Reply:
Good suggestion. I've renamed it.